<a href="https://colab.research.google.com/github/microize/datascience_portfolio/blob/main/0_project_4_basic_exploration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import numpy as np
import pandas as pd

pd.set_option('display.max_rows',None)
pd.set_option('display.max_columns',None)
pd.set_option('display.width',None)

In [11]:
df = pd.read_csv('/content/drive/MyDrive/Edvancer/Data/Project 4/Hazard_train.csv',index_col='Id')
test = pd.read_csv('/content/drive/MyDrive/Edvancer/Data/Project 4/Hazard_test_share.csv',index_col='Id')

### Sample Data

In [10]:
df.head(2)

Hazard  T1_V1  T1_V2  T1_V3 T1_V4 T1_V5 T1_V6 T1_V7 T1_V8 T1_V9  \
Id                                                                       
83642       2     10     23      4     N     K     Y     B     D     E   
92530       6      3     18      2     B     H     N     B     B     D   

       T1_V10 T1_V11 T1_V12  T1_V13  T1_V14 T1_V15 T1_V16 T1_V17  T2_V1  \
Id                                                                        
83642       2      H      B      10       0      A      K      N     52   
92530       8      L      B      10       3      A      B      N     39   

       T2_V2 T2_V3  T2_V4 T2_V5  T2_V6  T2_V7  T2_V8  T2_V9  T2_V10 T2_V11  \
Id                                                                           
83642     14     N      8     C      2     37      1     11       2      Y   
92530     14     Y     13     A      2     34      1     12       7      Y   

      T2_V12 T2_V13  T2_V14  T2_V15  
Id                                   
83642      N      E       2       1  
92530      N      C       3       3

In [12]:
test.head(2)

T1_V1  T1_V2  T1_V3 T1_V4 T1_V5 T1_V6 T1_V7 T1_V8 T1_V9  T1_V10 T1_V11  \
Id                                                                              
67943      9     18      2     N     B     Y     B     C     D      12      B   
38287     12     20      5     N     C     Y     D     B     D      12      B   

      T1_V12  T1_V13  T1_V14 T1_V15 T1_V16 T1_V17  T2_V1  T2_V2 T2_V3  T2_V4  \
Id                                                                             
67943      B      20       1      A      D      Y     75     13     N      5   
38287      B      10       1      A      E      N     27      8     Y     15   

      T2_V5  T2_V6  T2_V7  T2_V8  T2_V9  T2_V10 T2_V11 T2_V12 T2_V13  T2_V14  \
Id                                                                             
67943     A      2     34      1     14       4      Y      N      E       3   
38287     B      1     40      1      9       3      Y      N      E       1   

       T2_V15  
Id             
67943       3  
38287       1

Observation
* all columns are masked

### About Data

In [14]:
print("Train dataset shape:",df.shape)
print("Test dataset shape:",test.shape)

Train dataset shape: (40799, 33)
Test dataset shape: (10200, 32)


In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 40799 entries, 83642 to 47488
Data columns (total 33 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Hazard  40799 non-null  int64 
 1   T1_V1   40799 non-null  int64 
 2   T1_V2   40799 non-null  int64 
 3   T1_V3   40799 non-null  int64 
 4   T1_V4   40799 non-null  object
 5   T1_V5   40799 non-null  object
 6   T1_V6   40799 non-null  object
 7   T1_V7   40799 non-null  object
 8   T1_V8   40799 non-null  object
 9   T1_V9   40799 non-null  object
 10  T1_V10  40799 non-null  int64 
 11  T1_V11  40799 non-null  object
 12  T1_V12  40799 non-null  object
 13  T1_V13  40799 non-null  int64 
 14  T1_V14  40799 non-null  int64 
 15  T1_V15  40799 non-null  object
 16  T1_V16  40799 non-null  object
 17  T1_V17  40799 non-null  object
 18  T2_V1   40799 non-null  int64 
 19  T2_V2   40799 non-null  int64 
 20  T2_V3   40799 non-null  object
 21  T2_V4   40799 non-null  int64 
 22  T2_V5   40799 non-

* Column dont have null values

In [29]:
numeric_cols = (df.select_dtypes(include=['int64', 'float64']).columns).tolist()

print("The Number of Numeric Columns:",len(numeric_cols))
print(numeric_cols)

The Number of Numeric Columns: 17
['Hazard', 'T1_V1', 'T1_V2', 'T1_V3', 'T1_V10', 'T1_V13', 'T1_V14', 'T2_V1', 'T2_V2', 'T2_V4', 'T2_V6', 'T2_V7', 'T2_V8', 'T2_V9', 'T2_V10', 'T2_V14', 'T2_V15']


In [30]:
string_cols = (df.select_dtypes(include=['object']).columns).tolist()

print("The Number of String Columns:",len(string_cols))
print(string_cols)

The Number of String Columns: 16
['T1_V4', 'T1_V5', 'T1_V6', 'T1_V7', 'T1_V8', 'T1_V9', 'T1_V11', 'T1_V12', 'T1_V15', 'T1_V16', 'T1_V17', 'T2_V3', 'T2_V5', 'T2_V11', 'T2_V12', 'T2_V13']


### Target Variable Distribution

In [34]:
value_counts_sorted = df['Hazard'].value_counts().sort_index()
print(value_counts_sorted)

1     15229
2      4939
3      1632
4      6561
5      2902
6      1114
7      2651
8      1434
9       594
10     1048
11      633
12      311
13      442
14      304
15      142
16      183
17      154
18       90
19      111
20       71
21       20
22       53
23       31
24       11
25       34
26       20
27        4
28       27
29       12
30        5
31        5
32        4
33        1
34        6
35        1
37        2
38        2
39        1
40        2
41        2
42        2
44        1
46        2
49        1
51        1
52        1
63        1
64        1
69        1
Name: Hazard, dtype: int64


Observation

* The target variable has a wide range of values, ranging from 1 to 69.
* The majority class is labeled as 1 with 15,229 occurrences, followed by classes 4 and 2 with 6,561 and 4,939 occurrences, respectively.
* Some classes have relatively low frequencies, such as classes 35, 51, 64, 33, which have only 1 occurrence each.
* There is a gradual decrease in the frequency of classes as the class label increases.

### Descriptive statistics

In [37]:
df.describe()

Hazard         T1_V1         T1_V2         T1_V3        T1_V10  \
count  40799.000000  40799.000000  40799.000000  40799.000000  40799.000000   
mean       4.009584      9.723645     12.849359      3.190691      7.025123   
std        4.007785      5.170715      6.257502      1.742044      3.596619   
min        1.000000      1.000000      1.000000      1.000000      2.000000   
25%        1.000000      6.000000      7.000000      2.000000      3.000000   
50%        3.000000      9.000000     14.000000      3.000000      8.000000   
75%        5.000000     14.000000     18.000000      4.000000     12.000000   
max       69.000000     19.000000     24.000000      9.000000     12.000000   

             T1_V13        T1_V14         T2_V1         T2_V2         T2_V4  \
count  40799.000000  40799.000000  40799.000000  40799.000000  40799.000000   
mean      14.009902      1.582416     57.568641     12.410451     10.251403   
std        4.649012      0.865121     23.461468      4.785406      4.845443   
min        5.000000      0.000000      1.000000      1.000000      1.000000   
25%       10.000000      1.000000     40.000000      9.000000      6.000000   
50%       15.000000      1.000000     55.000000     11.000000     10.000000   
75%       20.000000      2.000000     77.000000     15.000000     14.000000   
max       20.000000      4.000000    100.000000     39.000000     22.000000   

              T2_V6         T2_V7         T2_V8         T2_V9        T2_V10  \
count  40799.000000  40799.000000  40799.000000  40799.000000  40799.000000   
mean       1.947499     33.504620      1.031349     12.496973      4.489963   
std        0.795249      5.830512      0.191676      7.313552      1.899386   
min        1.000000     22.000000      1.000000      1.000000      1.000000   
25%        2.000000     31.000000      1.000000      6.000000      3.000000   
50%        2.000000     34.000000      1.000000     14.000000      4.000000   
75%        2.000000     40.000000      1.000000     18.000000      6.000000   
max        7.000000     40.000000      3.000000     25.000000      7.000000   

             T2_V14        T2_V15  
count  40799.000000  40799.000000  
mean       2.448344      3.482512  
std        1.257293      3.070911  
min        1.000000      1.000000  
25%        2.000000      1.000000  
50%        2.000000      2.000000  
75%        3.000000      5.000000  
max        7.000000     12.000000

In [38]:
df[string_cols].describe()

T1_V4  T1_V5  T1_V6  T1_V7  T1_V8  T1_V9 T1_V11 T1_V12 T1_V15 T1_V16  \
count   40799  40799  40799  40799  40799  40799  40799  40799  40799  40799   
unique      8     10      2      4      4      6     12      4      8     18   
top         N      K      N      B      B      D      B      B      A      I   
freq    20060  11305  22856  38422  37146  19110  13694  37521  36531   7517   

       T1_V17  T2_V3  T2_V5 T2_V11 T2_V12 T2_V13  
count   40799  40799  40799  40799  40799  40799  
unique      2      2      6      2      2      5  
top         N      N      A      Y      N      E  
freq    32950  27597  27093  29520  33373  22008

### Test Data Validation
- Validating test data with train data observations

In [35]:
test_numeric_cols = (test.select_dtypes(include=['int64', 'float64']).columns).tolist()

print("The Number of Numeric Columns:",len(test_numeric_cols))
print(test_numeric_cols)

The Number of Numeric Columns: 16
['T1_V1', 'T1_V2', 'T1_V3', 'T1_V10', 'T1_V13', 'T1_V14', 'T2_V1', 'T2_V2', 'T2_V4', 'T2_V6', 'T2_V7', 'T2_V8', 'T2_V9', 'T2_V10', 'T2_V14', 'T2_V15']


In [36]:
test_string_cols = (test.select_dtypes(include=['object']).columns).tolist()

print("The Number of Numeric Columns:",len(test_string_cols))
print(test_string_cols)

The Number of Numeric Columns: 16
['T1_V4', 'T1_V5', 'T1_V6', 'T1_V7', 'T1_V8', 'T1_V9', 'T1_V11', 'T1_V12', 'T1_V15', 'T1_V16', 'T1_V17', 'T2_V3', 'T2_V5', 'T2_V11', 'T2_V12', 'T2_V13']


<br>

### Final Observation
---

* Train dataset shape: (40799, 33)
* Test dataset shape: (10200, 32)
* Column dont have null values
* The Number of Numeric Columns: 17
* The Number of String Columns: 16
* The target variable has a wide range of values, ranging from 1 to 69.
* Target variable: unbalanced / rightskewed

* **Numeric Columns**:
> ['Hazard', 'T1_V1', 'T1_V2', 'T1_V3', 'T1_V10', 'T1_V13', 'T1_V14', 'T2_V1', 'T2_V2', 'T2_V4', 'T2_V6', 'T2_V7', 'T2_V8', 'T2_V9', 'T2_V10', 'T2_V14', 'T2_V15']

* **String Columns**:
> ['T1_V4', 'T1_V5', 'T1_V6', 'T1_V7', 'T1_V8', 'T1_V9', 'T1_V11', 'T1_V12', 'T1_V15', 'T1_V16', 'T1_V17', 'T2_V3', 'T2_V5', 'T2_V11', 'T2_V12', 'T2_V13']

